src/mercury-inventory/mercury-inventory-sample.yaml

```diff
@@ -1,11 +1,14 @@
 inventory:
-  service:
-    bind_url: tcp://0.0.0.0:9000
+  bind_address: tcp://0.0.0.0:9000
```

Comment:
can we templatize this? e.g with Jinja2 (and others?) `tcp://{{inventory_bind_address}}:{{inventory_bind_port}}` instead? Would make dockerizing a lot easier since the docker container will be named `mercury_inventory`.

Reply:
For docker, wouldn't we ditch these configuration files and just use INVENTORY_SERVICE_BIND_ADDRESS environment variable?

Reply:
Also, this is a sample configuration file shipped with the source.. meant to be self documenting

Reply:
I'm mixed - I generally avoid ENV vars b/c they're leaky and much prefer using config files. OTOH, I'll basically be using ENV vars to setup the template with docker-compose any way...however, I'd only be setting a small handful of values and leaving the rest set via a script. `inventory_bind_address` will be based on the container name, so I can script that without an ENV var.

Q: If a config file is provided, does it override env/command-line? What is the order-of-precedence? And how many different settings are there? I'd rather not have 50+ env vars to manage settings.

I basically need to setup the logging, and then set the service names since `localhost` and `0.0.0.0` won't be used. Most everything else will be left to defaults.

Reply:
Oh, I see what you are saying now.. Yes url's could certainly be broken up. As for bringing in a template engine that might be overkill. Perhaps changing the config to:

```
service:
  bind_proto: tcp
  bind_interface: 0.0.0.0
  bind_port: 9000

--

service:
  bind_proto: icp
  bind_path: /var/run/sockets/inv

```
hmm.. documenting how to specify zeromq bind 'urls' might be the better solution.. (on the fence)

Reply:
Order of precedence is (greatest to least) command line > environment variable > configuration file

Reply:
okay - yeah; just commented on this config - but really, all of the config files are what I'm thinking of  this was just one example.